### PR TITLE
fix(connlib): remove outdated assertion due to possibility of network changes

### PR DESCRIPTION
**Pull request by conectado**
found while working on #4163

### CODE REVIEW

File: rust/connlib/snownet/src/node.rs [hunk 1]
```diff
@@ -144,6 +144,8 @@ where
         for allocation in self.allocations.values_mut() {
             allocation.refresh(now);
         }
+
+        self.bindings_and_allocations_drain_events();
```
Comment (thomaseizinger) [inline rust/connlib/snownet/src/node.rs]: This is actually wrong. `refresh` is async so we don't have any events here. But we should add this function to the top of `poll_event` to ensure we drain them in time.

**Review by thomaseizinger — CHANGES_REQUESTED** [review]

File: rust/connlib/snownet/src/node.rs (hunk 1 above)
Comment (conectado) [reply rust/connlib/snownet/src/node.rs]: Should I do it with this PR?

Comment (thomaseizinger) [reply rust/connlib/snownet/src/node.rs]: If you want to. I just don't want to add this here because it is pointless :)

Comment (conectado) [reply rust/connlib/snownet/src/node.rs]: added :)

**Comment by thomaseizinger** [comment]
@conectado That seems to break the snownet tests as those are very strict on what gets emitted when. I reckon just remove it for now. We might remove those tests soon and I don't want to block this on it.

Also, your PR title needs to be updated to match the implementation of the PR.

**Review by thomaseizinger — APPROVED** [review]
ACK modulo CI

**Pull request merged**